gateway-js/src/index.ts

```diff
@@ -116,11 +120,41 @@ function isCsdlConfig(config: GatewayConfig): config is CsdlGatewayConfig {
   return 'csdl' in config;
 }
 
+// A manually managed config means the user has provided a function which
+// handles providing service definitions to the gateway.
+function isManuallyManagedConfig(
+  config: GatewayConfig,
+): config is ManuallyManagedGatewayConfig {
+  return 'experimental_updateServiceDefinitions' in config;
+}
+
+// Managed config strictly means managed by Studio
 function isManagedConfig(
   config: GatewayConfig,
 ): config is ManagedGatewayConfig {
   return (
-    !isRemoteConfig(config) && !isLocalConfig(config) && !isCsdlConfig(config)
+    !isRemoteConfig(config) &&
+    !isLocalConfig(config) &&
+    !isCsdlConfig(config) &&
+    !isManuallyManagedConfig(config)
+  );
+}
+
+// A static config is one which loads synchronously on start and never updates
+function isStaticConfig(
```

Comment:
Would it be helpful to introduce StaticConfig and DynamicConfig interfaces that sit between LocalGatewayConfig (etc) and GatewayConfigBase, so that instead of returning `is LocalGatewayConfig | CsdlGatewayConfig` this returns `is StaticConfig`? (My grasp on TypeScript isn't strong enough to know if I'm suggesting an intermediate interface, or just `type StaticConfig = LocalGatewayConfig | CsdlGatewayConfig`.)

Reply:
Yep I really like this suggestion!

Reply:
af14289